clang/test/3C/root_cause.c

```diff
@@ -1,4 +1,5 @@
 // RUN: 3c -use-malloc=my_malloc -base-dir=%S -alltypes -warn-all-root-cause %s -- -Xclang -verify -Wno-everything
+// RUN: 3c -use-malloc=my_malloc -base-dir=%S -alltypes -warn-root-cause %s -- -Xclang -verify -Wno-everything
```

Comment:
Can you please add a test that `-warn-root-cause` still hides root causes that affect zero relevant pointers?  (I know this isn't a new problem in this PR, but this would be a good time to address it.)  If you add such a root cause, you should be able to add this test without duplicating all the other expected-diagnostic comments by having the `-warn-all-root-cause` command use `-verify` with multiple prefixes, much like we do with FileCheck: see [the documentation](https://clang.llvm.org/doxygen/classclang_1_1VerifyDiagnosticConsumer.html#details).